src/components/general/DatePicker.js

```diff
@@ -15,99 +16,76 @@ class DatePicker extends Component {
+		const maxDayCheck = (maxDay) => { if (day > maxDay) this.setState({ day: maxDay }); };
 
-		update(item) {
-			const {
-				month,
-				day,
-				year
-			} = item;
-			this.props.onSelect(`${year}-${month}-${day}`);
+		if (parseInt(month) == 2 && this.isLeapYear(year)) {
+			maxDayCheck(29);
+			return Array.from({ length: 29 }, (v, k) => k + 1);
 		}
-
-		clickActionMonth(item) {
-			const {
-				day,
-				year
-			} = this.state;
-
-			item = this.prepend0(item);
-
-			this.setState({ month: item });
-			const month30 = [false, false, false, true, false, true,
-				false, false, true, false, true, false];
-
-			if (item === 2)
-				if (year != 0 && this.leapYear(year)) {
-					this.setState({ dayArr: Array.from({ length: 29 }, (v, k) => k + 1) });
-					if (day > 29) this.setState({ day: 29 });
-				}
-				else {
-					this.setState({ dayArr: Array.from({ length: 28 }, (v, k) => k + 1) });
-					if (day > 28) this.setState({ day: 28 });
-				}
-			if (month30[item]) {
-				this.setState({ dayArr: Array.from({ length: 30 }, (v, k) => k + 1) });
-				if (day > 30) this.setState({ day: 30 });
-			}
-			else if (item !== 2) {
-				this.setState({ dayArr: Array.from({ length: 31 }, (v, k) => k + 1) });
-			}
-			if (day !== "" && year !== "")
-				this.update({ day: day, month: item, year: year });
+		else {
+			maxDayCheck(daysInMonth[parseInt(month) - 1]);
+			return Array.from({ length: daysInMonth[parseInt(month) - 1] }, (v, k) => k + 1);
 		}
-		clickActionDay(item) {
-			const {
-				month,
-				year
-			} = this.state;
+	}
 
-			item = this.prepend0(item);
-			this.setState({ day: item });
+	// Returns true if the input year is a leap year
+	isLeapYear(year) {
+		return year % 4 == 0 && year % 100 != 0 || year % 400 == 0;
+	}
 
-			if (month !== "" && year !== "")
-				this.update({ day: item, month: month, year: year });
-		}
+	update(item) {
+		const {
+			month,
+			day,
+			year
+		} = item;
+		this.props.onSelect(`${year}-${month}-${day}`);
+	}
 
-		clickActionYear(item) {
-			const {
-				month,
-				day
-			} = this.state;
+	// Calls the update function and sets M/D/Y based on a given value and type
+	clickAction(item, type) {
+		const {
+			month,
+			day,
+			year
+		} = this.state;
 
-			this.setState({ year: item });
+		item = prepend0(item);
 
-			if (month === "2" && this.leapYear(item)) {
-				this.setState({ dayArr: Array.from({ length: 29 }, (v, k) => k + 1) });
-				if (day > 29) this.setState({ day: "29" });
-			}
-			if (month !== "" && day !== "")
-				this.update({ day: day, month: month, year: item });
+		switch (type) {
+			case "month": this.setState({ month: item }); break;
+			case "day": this.setState({ day: item }); break;
+			case "year": this.setState({ year: item }); break;
 		}
 
+		if (month !== "" && day !== "" && year !== "") this.update({ day: day, month: month, year: year });
+	}
+
 	_keyExtractor = (item, index) => index;
 
 	render = () => {
```

Comment:
Although you didn't change this, can you make all of the destructuring statements one line?